plone/app/upgrade/v52/final.py

```diff
@@ -27,3 +27,33 @@ def rebuild_redirections(context):
     )
     storage._rebuild()
     logger.info('Done rebuilding redirections.')
+
+
+def move_dotted_to_named_behaviors(context):
+    """named behaviors are better then dotted behaviors > let's move them."""
+    from plone import api
+    from plone.behavior.registration import lookup_behavior_registration
+    from plone.dexterity.interfaces import IDexterityFTI
+
+    ptt = api.portal.get_tool('portal_types')
+
+    ftis = [fti for fti in ptt.objectValues() if IDexterityFTI.providedBy(fti)]
+
+    for fti in ftis:
+        behaviors = []
+        for behavior in fti.behaviors:
+            behavior_registration = lookup_behavior_registration(behavior)
+            behaviors.append(behavior_registration.name)
```

Comment:
Off-head: I think it is still possible to register behaviors w/o a name. In this case None would be appended.

Reply:
good point - i did not take that into consideration. i will make that less strict